### DISCUSSION TITLE
Puzzle 18 Seating Chart

### DISCUSSION THREAD

**Discussion by maximilian22x** [smpetersgithub/AdvancedSQLPuzzles#13] [General]
https://github.com/smpetersgithub/AdvancedSQLPuzzles/blob/66533d023b70ef3d234a302a2453951bee29936f/Advanced%20SQL%20Puzzles/Advanced%20SQL%20Puzzles%20Solutions.sql#L828

** Also possible, but your version is better.**

WITH cte_Gaps AS 
(
SELECT  SeatNumber                                                   AS GapStart,
        LEAD(SeatNumber,1,0) OVER (ORDER BY SeatNumber)              AS GapEnd,
        LEAD(SeatNumber,1,0) OVER (ORDER BY SeatNumber) - SeatNumber AS Gap
FROM    #SeatingChart
)
SELECT  sum((GapEnd - 1 )  - (GapStart + 1 )  + 1)
FROM    cte_Gaps
WHERE Gap > 1;

GO

**Comment by smpetersgithub** [comment]
Thank you, I will check out this syntax.  This puzzle was originally in Joe Celko's SQL Puzzle and Answer book, which is where I learned how to solve this puzzle.

Thank you for letting me know of all the small errors in the puzzles.  Really appreciate it.  Please keep them coming.

**Comment by smpetersgithub** [comment]
Closing
